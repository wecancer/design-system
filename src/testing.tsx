import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
