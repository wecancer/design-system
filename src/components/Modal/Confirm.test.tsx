import ModalConfirm from './Confirm'
import {render} from '../../testing'

describe('<ModalConfirm />', () => {
  beforeEach(() => {
    if (!document.getElementById('root')) {
      const root = document.createElement('div')
      root.setAttribute('id', 'root')
      document.body.appendChild(root)
    }
  })

  it('should render snapshot correctly', () => {
    const {container} = render(
      <ModalConfirm cancelLabel="Cancel" confirmLabel="Confirm">
        Something
      </ModalConfirm>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
