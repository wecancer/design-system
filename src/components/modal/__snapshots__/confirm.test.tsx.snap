// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalConfirm /> should render snapshot correctly 1`] = `null`;
